src/ahp/prover.rs

```diff
@@ -294,7 +296,9 @@ impl<F: PrimeField> AHPForR1CS<F> {
             z_a: Some(z_a),
             z_b: Some(z_b),
             w_poly: None,
+            mz_rands: None,
             mz_polys: None,
+            mz_rand_polys: None,
```

Comment:
I don't think we need to store both `mz_polys` and `mz_rand_polys`; we can store just `mz_rand_polys`, and then later on, when we need to compute `z_a * z_b`, we just temporarily mutate `mz_rand_polys` in place by subtracting `mz_rands * v_H`.